### ISSUE TITLE
DDIMScheduler silently breaking when using epsilon prediction and rescaling betas to zero SNR

### ISSUE THREAD

**Issue by f90** [huggingface/diffusers#3851]
### Describe the bug

When using the `DDIMScheduler` with arguments `prediction_type="epsilon"` (which is default) and activating `rescale_betas_zero_snr=True`, and then trying to start sampling with it, one encounters `Inf` values in the returned prediction immediately at the first step. 

This occurs because of [this line](https://github.com/huggingface/diffusers/blob/0bab447670f47c28df60fbd2f6a0f833f75a16f5/src/diffusers/schedulers/scheduling_ddim.py#L390) dividing by zero - but I don't think there is an obvious fix to that line that makes sense.

The [paper](https://arxiv.org/pdf/2305.08891.pdf) proposing the beta rescaling switches to v-diffusion (`prediction_type="v_prediction"`), and when I do that with the `DDIMScheduler` it also works again normally. The paper doesn't explicitly state that inference is not possible with epsilon prediction, only that the model predicts something that is not very useful. Given the sampling equations however, it does look like the rescaling is just not usable with epsilon prediction. So probably the best solution is to just error out to the user in this case to state that these two settings are incompatible with each other.

`clip_sample=True` hides the issue by instead inserting the min value into the tensor, but the overall problem remains.

### Reproduction

import torch
from diffusers import DDIMScheduler
scheduler = DDIMScheduler(1000, clip_sample=False, rescale_betas_zero_snr=True)
scheduler.set_timesteps(50)
res = scheduler.step(model_pred, 999, curr_sample)
assert(torch.all(torch.isfinite(res.prev_sample)))

### System Info

- `diffusers` version: 0.17.1
- Platform: macOS-13.4-x86_64-i386-64bit
- Python version: 3.8.9
- PyTorch version (GPU?): 1.12.0 (False)
- Huggingface_hub version: 0.15.1
- Transformers version: 4.25.1
- Accelerate version: 0.19.0
- xFormers version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@patrickvonplaten @williamberman

**Comment by patrickvonplaten** [comment]
Hey @f90,

Can you please add a reproducible code snippet?

**Comment by patrickvonplaten** [comment]
I cannot run:

```py
import torch
from diffusers import DDIMScheduler
scheduler = DDIMScheduler(1000, clip_sample=False, rescale_betas_zero_snr=True)
scheduler.set_timesteps(50)
res = scheduler.step(model_pred, 999, curr_sample)
assert(torch.all(torch.isfinite(res.prev_sample)))
```

As I don't know what `model_pred` is. Would be great if you could add a complete code snippet :-)

**Comment by f90** [comment]
Sorry, my bad, here is the reproducible snippet I was able to run on my end just now:
```
import torch
from diffusers import DDIMScheduler
scheduler = DDIMScheduler(1000, clip_sample=False, rescale_betas_zero_snr=True)
scheduler.set_timesteps(50)
curr_sample = torch.randn(8, 100)
model_pred = torch.randn_like(curr_sample)
res = scheduler.step(model_pred, 999, curr_sample)
assert(torch.all(torch.isfinite(res.prev_sample)))
```
I think `curr_sample` and `model_pred` can be of any value and shape that you like, it will always break because of a division by zero.

**Comment by f90** [comment]
That's a good point. I forgot to mention that this normally only arises when then also using `timestep_spacing="trailing"` as this shifts the timesteps to start at the very first step (999 in our case), so there it always breaks:

```
import torch
from diffusers import DDIMScheduler
scheduler = DDIMScheduler(1000, clip_sample=False, rescale_betas_zero_snr=True, 
                          timestep_spacing="trailing")
scheduler.set_timesteps(50)
curr_sample = torch.randn(8, 100)
model_pred = torch.randn_like(curr_sample)
res = scheduler.step(model_pred, scheduler.timesteps[0], curr_sample)
assert(torch.all(torch.isfinite(res.prev_sample)))
```

This problem also occurs when using all timesteps at inference:
```
import torch
from diffusers import DDIMScheduler
scheduler = DDIMScheduler(1000, clip_sample=False, rescale_betas_zero_snr=True)
scheduler.set_timesteps(1000)
curr_sample = torch.randn(8, 100)
model_pred = torch.randn_like(curr_sample)
res = scheduler.step(model_pred, scheduler.timesteps[0], curr_sample)
assert(torch.all(torch.isfinite(res.prev_sample)))
```
Although it's more of a rare scenario to use DDIM at inference time with the same number of sampling steps as during inference but probably still something that people want to do. 

So basically we should throw a special error when
1) doing epsilon prediction and
2) rescaling betas and
3) `scheduler.timesteps[0]` is used for inference which could be due to trailing timesteps setting or `scheduler.set_timesteps(N)` sets an `N` so high so that it becomes the very first step

**Comment by patrickvonplaten** [comment]
Definitely interested in better error messages - feel free to open a PR if you think it helps the UX
